activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -651,7 +651,7 @@ def exec_cache(sql, name, binds)
         CACHED_PLAN_HEURISTIC = "cached plan must not change result type".freeze
         def is_cached_plan_failure?(e)
           pgerror = e.cause
-          code = pgerror.result.result_error_field(PGresult::PG_DIAG_SQLSTATE)
+          code = pgerror.result.result_error_field(PG::PG_DIAG_SQLSTATE)
```

Comment:
`PGresult` has been renamed to `PG::Result` [source](https://github.com/ged/ruby-pg/blob/master/History.rdoc#v0130-2012-02-09-michael-granger-gedfaeriemudorg)